nixos/modules/services/hardware/auto-epp.nix

```
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+
+ cfg = config.services.auto-epp;
+ package = pkgs.auto-epp;
+
+in {
+ options = {
+
+    services.auto-epp = {
+      enable = mkEnableOption (lib.mdDoc "auto-epp for amd active pstate");
+      batState= mkOption {
+        type = types.str;
+        default = "power";
+        description = lib.mdDoc "energy_performance_preference when on battery";
+      };
+
+     acState= mkOption {
+        type = types.str;
+        default = "balance_performance";
+        description = lib.mdDoc "energy_performance_preference when plugged in";
+      };
```

Although there are just 2 settings at the moment, I'd still recommend following RFC 42 with `format = pkgs.formats.ini { };`

```suggestion
      settings = types.submodule {
        freeformType = format.type;
        options = {
          settings = {
            epp_state_for_AC = mkOption {
              type = types.str;
              default = "power";
              description = lib.mdDoc ''
                energy_performance_preference when on battery

								::: {.note}
								See available epp states by running:
								{command}`cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences`
								:::
              '';
            };

            epp_state_for_BAT = mkOption {
              type = types.str;
              default = "balance_performance";
              description = lib.mdDoc ''
                `energy_performance_preference` when plugged in

								::: {.note}
								See available epp states by running:
								{command}`cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences`
								:::
              '';
            };
          };
        };
      };
```

---

I tried this on my machines and it is not working, i tried the following:
```nix
services.auto-epp = {
     enable = true;
     settings.epp_state_for_AC = "performance";
  };
# or
services.auto-epp = {
     enable = true;
     settings.settings.epp_state_for_AC = "performance";
  };
# or
services.auto-epp = {
     enable = true;
     settings.epp_state_for_AC = "performance";
  };
```
I always get that the option does not exist. 

I managed to make it work with:
```nix
services.auto-epp = {
      enable = mkEnableOption (lib.mdDoc "auto-epp for amd active pstate");
     # settings = lib.mkOption {
        type = lib.types.submodule {
          freeformType = format.type;
          options = {
            settings.epp_state_for_AC = mkOption {
        ....
```

but then I need  to set i t up this way:
```nix
services.auto-epp = {
     enable = true;
     settings.settings.epp_state_for_AC = "performance";
  };
```

which looks silly